core/unit_test/TestReduce.hpp

```diff
@@ -625,4 +625,42 @@ TEST(TEST_CATEGORY, int_combined_reduce_mixed) {
 }
 #endif
 #endif
+
+#if defined(NDEBUG)
+template <class T>
+struct EnableTestForIssue6517 : std::false_type {};
+
+#ifdef KOKKOS_ENABLE_CUDA
+template <>
+struct EnableTestForIssue6517<Kokkos::Cuda> : std::true_type {};
+#endif
+#ifdef KOKKOS_ENABLE_HIP
+template <>
+struct EnableTestForIssue6517<Kokkos::HIP> : std::true_type {};
+#endif
+#ifdef KOKKOS_ENABLE_SYCL
+template <>
+struct EnableTestForIssue6517<Kokkos::Experimental::SYCL> : std::true_type {};
+#endif
+
+struct FunctorIssue6517 {
+  KOKKOS_FUNCTION void operator()(const int64_t /*i*/, double& update) const {
+    update += 1.0;
+  }
+};
+
+TEST(TEST_CATEGORY, issue6517) {
```

Comment:
Can we find a better, more descriptive name than `issue617`?

Reply:
i like it because we can easily find the issue :) 
you have any suggestions?

Reply:
`large_iteration_count` or something like that.